### PR TITLE
R1 WYSIWYG Panel

### PR DESCRIPTION
**Pull request by jamiepaul**
Ticket: https://central.tri.be/issues/131670
Design: https://moderntribe.invisionapp.com/d/main#/console/16185992/347968146/preview

Adds layout options to align panel header (left/center), with center being the default as this was previously the default in [1]. Implements design for all screen sizes.

### CODE REVIEW

File: wp-content/plugins/core/src/Panels/Types/Wysiwyg.php [hunk 1]
```diff
@@ -12,6 +12,10 @@ class Wysiwyg extends Panel_Type_Config {
 	const FIELD_DESCRIPTION    = 'description';
 	const FIELD_COLUMNS        = 'columns';
 	const FIELD_COLUMN_CONTENT = 'column_content';
+	const FIELD_LAYOUT         = 'layout';
+
+	const OPTION_LAYOUT_LEFT   = 'option_wysiwyg_layout_left';
```
Comment (ryanurban) [inline wp-content/plugins/core/src/Panels/Types/Wysiwyg.php]: See here: https://github.com/moderntribe/square-one/pull/355#discussion_r313149269 and how that flows into the wrapper layout class ;)

File: wp-content/themes/core/pcss/content/panel/_wysiwyg.pcss [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		}
+	}
+
+	.g-col:not(:last-child) {
+		margin-bottom: var(--spacer-4);
+	}
+}
+
+.s-header {
+	@media (--viewport-medium) {
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/content/panel/_wysiwyg.pcss]: Remember that this is a globally available section utility class, so we should stay aware of whether the change we are making should be applicable only to this particular panel (sandbox it in your panel class) or is a global change (port to the sections header partial) and handle accordingly ;)

Fwiw, these do look like global changes.

File: wp-content/plugins/core/src/Panels/Types/Wysiwyg.php (hunk 1 above)
Comment (jamiepaul) [reply wp-content/plugins/core/src/Panels/Types/Wysiwyg.php]: Updated

File: wp-content/themes/core/pcss/content/panel/_wysiwyg.pcss (hunk 2 above)
Comment (jamiepaul) [reply wp-content/themes/core/pcss/content/panel/_wysiwyg.pcss]: My mistake. I'm not sure the width should be global for this, so scoping to the panel until we see that pattern really emerge.

File: wp-content/plugins/core/src/Templates/Content/Panels/Wysiwyg.php [hunk 3]
```diff
@@ -14,15 +20,64 @@ public function get_data(): array {
...
+
+	/**
+	 * Get WYSIWYG row wrapper classes.
+	 *
+	 * @return string
+	 */
+	protected function get_wrapper_classes() {
+		$classes = [];
+		$columns = $this->panel_vars[ Wysi::FIELD_COLUMNS ];
+
```
Comment (speerface) [inline wp-content/plugins/core/src/Templates/Content/Panels/Wysiwyg.php]: Let's check if $columns is empty here and return `Util::class_attribute( $classes, false )` if so. Then we can continue without having to check that in the next to conditionals.

**Comment by speerface** [comment]
Let's make sure to add in a changelog entry for this too :)

File: wp-content/themes/core/content/panels/wysiwyg.twig [hunk 4]
```diff
@@ -2,10 +2,26 @@
 
 {% extends "content/panels/panel.twig" %}
 
+{% block header %}
```
[the remark this answers is not in the document]
Comment (jamiepaul) [reply wp-content/themes/core/content/panels/wysiwyg.twig]: Indeed. Good catch!

File: wp-content/plugins/core/src/Templates/Content/Panels/Wysiwyg.php (hunk 3 above)
Comment (ryanurban) [reply wp-content/plugins/core/src/Templates/Content/Panels/Wysiwyg.php]: @speerface is that needed? Saw this on the project where it's getting implemented and seems like it's redundant given you still end up with the same return statement.

https://github.com/moderntribe/wstc/pull/1#discussion_r313949859

**Pull request merged**